cloud/migrator.js

```diff
@@ -221,6 +225,67 @@ Migrator.prototype.getAfterDelete = function(klass) {
   }
 };
 
+// Get bulk import returns a function that runs either the user's explicit
+// importer or the migration function. Any changes made by the import function
+// plus a change to the object's migration state are applied to the objects,
+// which are saved back in bulk.
+// Returns the number of objects imported.
+Migrator.prototype.getBulkImport = function(klass) {
+  var migrate = this.handlers(klass).migrateObject,
+    bulkImport = this.handlers(klass).bulkImport,
+      _Promise = this._parse.Promise;
+      _Parse = this._parse;
+
+  if (_.isUndefined(migrate) && _.isUndefined(bulkImport)) {
+    return undefined;
+  }
+
+  return function(objects) {
+    var migrations;
+    if (_.isUndefined(bulkImport)) {
+      // If the user has not defined a bulkImport method, we first let them run their
+      // migration methods in parallel. After they all complete, we batch the saves
+      // in a Parse.Object.saveAll request. We sequence these methods because saveAll
+      // counts in Parse as N requests and a parallel save is likely to make customers
+      // hit their throughput limit.
+      migrations = _.map(objects, function(object) {
+        return _Promise.as().then(function() {
+          return migrate(object);
+        }).then(function(maybeChanged) {
+          var ret = maybeChanged instanceof _Parse.Object ? maybeChanged : object;
+          ret.set(consts.MIGRATION_KEY, consts.IMPORTED);
+          return ret;
+        });
+      });
+    } else {
+      migrations = bulkImport(objects).then(function(changed) {
+        return _.map(changed, function(object) {
+          object.set(consts.MIGRATION_KEY, consts.IMPORTED);
+        });
+      });
+    }
+
+    return _Promise.when(migrations).then(function(objects) {
+      var saveBack = _Promise.as();
+      for (var start = 0; start < objects.length; start += consts.SAVE_BATCH_SIZE) {
+        var slice = objects.slice(start, start + consts.SAVE_BATCH_SIZE);
+        saveBack = saveBack.then(function() {
+          var p = new _Promise();
+          _Parse.Object.saveAll(slice, {
```

Comment:
It looks to me like saveAll already returns a Promise, at least in https://www.parsecdn.com/js/parse-1.5.0.js

If there's some other reason for this, please add a comment.